### DISCUSSION TITLE
Which MapLibre Style elements are supported by the Node.js platform?

### DISCUSSION THREAD

**Discussion by jechterhoff** [maplibre/maplibre-native#1786] [Q&A]
Hi.

I'm struggling to find information about which of the [MapLibre Style](https://maplibre.org/maplibre-style-spec/) elements are actually supported by the maplibre-native Node.js platform. Is there a suitable (easy?) way to figure this out?

https://www.npmjs.com/package/@maplibre/maplibre-gl-native shows release version 5.2.0 (which a [search in the maplibre-native releases](https://github.com/maplibre/maplibre-native/releases?q=node&expanded=true) also digs up). When testing a particular style that I wrote - and which works with MapLibre GL JS - I found out the hard way (through trial and error) that [icon-overlap](https://maplibre.org/maplibre-style-spec/layers/#layout-symbol-icon-overlap) was not supported by the Node.js platform.

The SDK support table for `icon-overlap` in the Style Spec lists `MapLibre GL JS`, `Android SDK`, `iOS SDK`, and `macOS SDK`:

![grafik](https://github.com/maplibre/maplibre-native/assets/4831647/c7b1fc25-1ece-4cef-9534-3eabf6e3380a)

For the latter three, "Not yet supported" is shown, so it is fair to assume that Node.js also does not support `icon-overlap`. That should be easy to deduce. But is there a particular SDK (out of the four which are listed in the MapLibre Style Spec) that I should be looking at in order to determine what is supported by the Node.js platform? Like: "Look at what is supported by the Android SDK, for version 5.2.0". That would be valuable information for reviewing other style elements, regarding support in the Node.js platform.

Apologies if I'm missing an important point here. I'm just struggling to find out which MapLibre Style elements are and are not supported by the maplibre-native Node.js platform release 5.2.0.

Thanks in advance for any help or advice you can give on this topic.

**Comment by louwers** [comment]
This is very useful feedback! It seems that we need to extend the version matrix to include Node.js as well.

Let's create an issue over at [maplibre-style-spec](https://github.com/maplibre/maplibre-style-spec/) for this.

**Reply by ntadej** [reply]
This is why it may be useful to version native core separately, because in principle all Native platforms share the same style spec support, just the versioning is out of sync.

**Reply by louwers** [reply]
Yes I think a versioned core is a good idea. But I think documenting the support should maybe just be a boolean (core support or not) to simplify things.

I wrote a proposal here: https://github.com/maplibre/maplibre-style-spec/issues/391
